### DISCUSSION TITLE
Failed to enable PGSODIUM Extension: function pgsodium.crypto_aead_det_decrypt(bytea, bytea, uuid, bytea) does not exist

### DISCUSSION THREAD

**Discussion by kjell0w** [supabase/supabase#16835] [Contribute to Supabase]
# Bug report

## Describe the bug

When trying to enable the PGSODIUM Extension in the Supabase Dashboard, it throws this Error "Failed to toggle PGSODIUM: failed to create pg.extensions: function pgsodium.crypto_aead_det_decrypt(bytea, bytea, uuid, bytea) does not exist".
I've tried to enable it for the extensions and pgsodium schema and both failed with the same error message.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to 'Dashboard / Database / Extensions'
2. Find 'PGSODIUM'
3. Click the Toggle to enable it
4. Choose pgsodium or extensions schema
5. Click 'Enable addon'

## Expected behavior

I expected the Extension to be enabled.

## Screenshots

![2023-08-18-102713_hyprshot](https://github.com/supabase/supabase/assets/55231995/6ead478b-1e92-4190-a5ff-32edc4499aed)

![2023-08-18-102709_hyprshot](https://github.com/supabase/supabase/assets/55231995/c09c0497-f49b-4362-b7d6-16d12189fdf9)

## System information

- OS: Linux
- Browser (if applies) [e.g. chrome, safari]: Brave
- Version of supabase-js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

**Comment by monicakh** [comment]
This error might be due to an old version of pgsodium. You can try to upgrade your project from [here](https://supabase.com/dashboard/project/_/settings/infrastructure) to upgrade your project to the latest PG15.
